pilot/test/integration/main.go

```diff
@@ -0,0 +1,107 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"flag"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	params = newInfra()
+
+	// Enable/disable auth, or run both for the tests.
+	authmode string
+	verbose  bool
+)
+
+func init() {
+	flag.StringVar(&params.Hub, "hub", params.Hub, "Docker hub")
+	flag.StringVar(&params.Tag, "tag", params.Tag, "Docker tag")
+	flag.StringVar(&params.IstioNamespace, "ns", params.IstioNamespace,
+		"Namespace in which to install Istio components (empty to create/delete temporary one)")
+	flag.StringVar(&params.Namespace, "n", params.Namespace,
+		"Namespace in which to install the applications (empty to create/delete temporary one)")
+	flag.StringVar(&params.Registry, "registry", params.Registry, "Pilot registry")
+	flag.BoolVar(&verbose, "verbose", false, "Debug level noise from proxies")
+	flag.BoolVar(&params.CheckLogs, "logs", params.CheckLogs,
+		"Validate pod logs (expensive in long-running tests)")
+
+	flag.StringVar(&params.KubeConfig, "kubeconfig", params.KubeConfig,
+		"kube config file (missing or empty file makes the test use in-cluster kube config instead)")
+	flag.IntVar(&params.TestCount, "count", params.TestCount, "Number of times to run each test")
+	flag.StringVar(&authmode, "auth", "both", "Enable / disable auth, or test both.")
+	flag.BoolVar(&params.Mixer, "mixer", params.Mixer, "Enable / disable mixer.")
+	flag.StringVar(&params.ErrorLogsDir, "errorlogsdir", params.ErrorLogsDir,
+		"Store per pod logs as individual files in specific directory instead of writing to stderr.")
+	flag.StringVar(&params.CoreFilesDir, "core-files-dir", params.CoreFilesDir,
+		"Copy core files to this directory on the Kubernetes node machine.")
+
+	// If specified, only run one test
+	flag.StringVar(&params.SelectedTest, "testtype", params.SelectedTest,
+		"Select test to run (default is all tests)")
+
+	flag.BoolVar(&params.UseAutomaticInjection, "use-sidecar-injector", params.UseAutomaticInjection,
+		"Use automatic sidecar injector")
+	flag.BoolVar(&params.UseAdmissionWebhook, "use-admission-webhook", params.UseAdmissionWebhook,
+		"Use k8s external admission webhook for config validation")
+
+	flag.StringVar(&params.AdmissionServiceName, "admission-service-name", params.AdmissionServiceName,
+		"Name of admission webhook service name")
+
+	flag.IntVar(&params.DebugPort, "debugport", params.DebugPort, "Debugging port")
+
+	flag.BoolVar(&params.debugImagesAndMode, "debug", params.debugImagesAndMode,
+		"Use debug images and mode (false for prod)")
+	flag.BoolVar(&params.SkipCleanup, "skip-cleanup", params.SkipCleanup,
+		"Debug, skip clean up")
+	flag.BoolVar(&params.SkipCleanupOnFailure, "skip-cleanup-on-failure", params.SkipCleanupOnFailure,
+		"Debug, skip clean up on failure")
+}
+
+// Main entry point for integration test executable.
+func main() {
```

Comment:
Confused, if you do 'go test ..' - how do the flags get set ? 
Also, why do we still have the main - is it for transition ? Again, please add TODOs or indicate the intentions...

For new tests I would expect people to just write normal go test, and in time existing test would be converted - I don't mind a transition period, but having both main and test may confuse people.

Reply:
The point of this PR is to maintain what we have today, which is an executable that runs our integration tests.  The `main.go` simply moves the main function to its own file, this isn't new. 

What's new in this PR is `driver_test.go`, which is a golang test.  If you were to run `go test` (and had the appropriate environment variables set) this is what would be run (not `main`).

I expect that we'll eventually remove `main.go` in favor of `driver_test.go`, but we need to make sure that all uses of `main` are swapped over to use golang tests first.